esp-hal/src/interrupt/riscv.rs

```diff
@@ -37,6 +37,10 @@ pub enum Error {
     InvalidInterruptPriority,
     /// The CPU interrupt is a reserved interrupt
     CpuInterruptReserved,
+    /// The enabled trap mode is not correct
+    InvalidTrapMode,
```

Comment:
I don't think these two errors can be caused by the user - we should have asserts instead